src/bin/pgcopydb/catalog.c

```diff
@@ -6998,6 +6998,58 @@ catalog_upsert_process_info(DatabaseCatalog *catalog, ProcessInfo *ps)
 }
 
 
+/*
+ * catalog_delete_s_table deletes an s_table entry for the given oid.
+ */
+bool
+catalog_truncate_process(DatabaseCatalog *catalog)
```

Comment:
I think this can go in a separate PR. This is to fix the original issue discussed here in https://github.com/dimitri/pgcopydb/issues/692#issue-2167452958

Reply:
Yeah, it makes sense to isolate that parts in its own PR, please do!